### PR TITLE
[server/OAPIF] Add full pagination for collection items HTML page

### PR DESCRIPTION
**Pull request by pathmapper**
full_pagination

---

**Before this PR**

---
Also fixes the case when `limit=0`.

~~Temporarily includes #52010, I will rebase when that is merged.~~ -> done

### CODE REVIEW

File: src/server/services/wfs3/qgswfs3handlers.cpp [hunk 1]
```diff
@@ -1395,23 +1399,94 @@ void QgsWfs3CollectionsItemsHandler::handleRequest( const QgsServerApiContext &c
...
+          const std::string firstPageLink { cleanedUrlAsString.toStdString() + QStringLiteral( "offset=0&limit=%1" ).arg( limit ).toStdString() };
+          const std::string lastPageLink { cleanedUrlAsString.toStdString() + QStringLiteral( "offset=%1&limit=%2" ).arg( totalPages * limit - limit ).arg( limit ).toStdString() };
+
+          if ( currentPage != 1 )
+          {
+            pagination.push_back( {{ "title",  "1" }, { "href", firstPageLink }, { "class", "page-item" }} ) ;
+          }
+          if ( currentPage > 3 )
+          {
+            pagination.push_back( {{ "title", "..." }, { "class", "page-item disabled" }} ) ;
```
Comment (elpaso) [inline src/server/services/wfs3/qgswfs3handlers.cpp:1456]: Maybe use the unicode char for three dots: …

File: src/server/services/wfs3/qgswfs3handlers.cpp [hunk 2]
```diff
@@ -1395,23 +1399,94 @@ void QgsWfs3CollectionsItemsHandler::handleRequest( const QgsServerApiContext &c
...
+            pagination.push_back( {{ "title",  std::to_string( currentPage - 1 ) }, { "href", prevPageLink }, { "class", "page-item" }} ) ;
+          }
+          pagination.push_back( {{ "title",  std::to_string( currentPage ) }, { "href", currentPageLink }, { "class", "page-item active" }} ) ;
+          if ( currentPage < totalPages - 1 )
+          {
+            pagination.push_back( {{ "title",  std::to_string( currentPage + 1 )  }, { "href", nextPageLink }, { "class", "page-item" }} ) ;
+          }
+          if ( currentPage < totalPages - 2 )
+          {
+            pagination.push_back( {{ "title", "..." }, { "class", "page-item disabled" }} ) ;
```
Comment (elpaso) [inline src/server/services/wfs3/qgswfs3handlers.cpp:1469]: same

**Review by elpaso — APPROVED** [review]
Nice work!

File: src/server/services/wfs3/qgswfs3handlers.cpp (hunk 1 above)
Comment (pathmapper) [reply src/server/services/wfs3/qgswfs3handlers.cpp]: Good idea, addressed per https://github.com/qgis/QGIS/pull/52051/commits/8fc94611895dca931867e6748971c2bb69754fa4

File: src/server/services/wfs3/qgswfs3handlers.cpp (hunk 2 above)
Comment (pathmapper) [reply src/server/services/wfs3/qgswfs3handlers.cpp]: addressed per https://github.com/qgis/QGIS/pull/52051/commits/8fc94611895dca931867e6748971c2bb69754fa4

**Pull request merged**